### PR TITLE
Fix controller-tools module

### PR DESCRIPTION
**Pull request by christarazi**
See commit msgs.

### CODE REVIEW

**Comment by christarazi** [comment]
No need for the whole CI to run, as the GH Action checks for the `go.mod` and Travis is enough.

File: go.mod
Comment (sayboras) [inline go.mod:112]: :100: 

Just curious if we have any plan to move this to cilium org (similar to client-go), this will help to avoid non-existent hash issue to some capacity :).

Comment (christarazi) [reply go.mod:112]: Yes, I think in the near-future we should move this repo under the Cilium org. Once we merge in https://github.com/cilium/cilium/pull/11477 and finish all the followups (https://github.com/cilium/cilium/issues/12737), likely the controller-tools fork will stabilize and we can "promote" it to be under the Cilium org.

Edit: Added as an item under the followups issue mentioned above.

File: Makefile
Comment (glibsm) [inline Makefile:289]: is this line relevant?

Comment (christarazi) [reply Makefile:289]: Yes it is relevant because bumping to the latest controller-tools will cause v1 CRDs to be generated. Since we are still on v1beta1 CRDs, we need this line. The PR to migrate to v1 CRDs is https://github.com/cilium/cilium/pull/11477. (It is also the PR that I was working on where I likely messed up the SHAs due to force-pushing.)

**Review by rolinh — APPROVED** [review]

**Review by aanm — APPROVED** [review]
Why don't we need to regenerate the CRDs?

**Review by sayboras — APPROVED** [review]

**Comment by christarazi** [comment]
> Why don't we need to regenerate the CRDs?

Because nothing changed about the CRDs. The only reason for this PR is that the controller-tools fork was updated to default to generating v1 CRDs, and therefore we needed to change the `CRD_OPTIONS` in the Makefile to force v1beta1 generation.

**Comment by christarazi** [comment]
Code owner approval and related GH Action checks for `go.mod` and generating K8s API have passed; this is enough to mark ready to merge.

**Pull request merged**
